cli/azd/pkg/apphost/generate.go

```diff
@@ -828,9 +828,11 @@ func (b *infraGenerator) buildEnvBlock(env map[string]string, manifestCtx *genCo
 		// And YAML rules would treat the above as the value being a boolean instead of a string, which the container
 		// app service expects.
 		//
-		// JSON marshalling the string value will give us something like `"true"` (with the quotes, and any escaping
+		// YAML marshalling the string value will give us something like `"true"` (with the quotes, and any escaping
 		// that needs to be done), which is what we want here.
-		jsonStr, err := json.Marshal(res)
+		// Do not use JSON marshall as it would escape the quotes within the string, breaking the meaning of the value.
+		// yaml marshall will use 'some text "quoted" more text' as a valid yaml string.
+		jsonStr, err := yaml.Marshal(res)
```

Comment:
I'm trying to wrap my head around what this ends up being. `yaml.Marshal` generally would produce marshal the input into a YAML document. `json.Marshal` produces the same, but a single node is a valid JSON document..

What does `yaml.Marshal([]byte("true"))` produce?

Reply:
Ah, looks like `yaml.Marshal` accepts `string` as well, `yaml.Marshal("true")` returns the literal value `"true"`

Reply:
`yaml.Marshal` marshals the string-value as a YAML string, so this works as expected.